src/main/java/com/google/opensesame/projects/ProjectEntity.java

```diff
@@ -84,33 +65,28 @@ public ProjectEntity(
     }
   }
 
-  public int getNumMentors() {
-    return this.mentorIds.size();
+  /** Compute the lengths of mentor and interested user lists before the project entity is saved. */
+  @OnSave
+  protected void computeListLengths() {
+    numMentors = mentorIds.size();
+    numInterestedUsers = interestedUserIds.size();
   }
 
-  public int getNumInterestedUsers() {
-    return this.interestedUserIds.size();
+  @OnLoad
+  protected void checkNullLists() {
+    if (this.mentorIds == null) {
+      this.mentorIds = new ArrayList<String>();
+    }
+    if (this.interestedUserIds == null) {
+      this.interestedUserIds = new ArrayList<String>();
```

Comment:
I think Objectify throws an error if you save null. It might be worth it to use this check on save too.

Reply:
Oh it does? Let me verify that on my end, and in that case I'll add that to the `@OnSave`.

Reply:
I just tested that and it seems like Objectify isn't having any issues "storing" null values. This corresponds with the idea that Objectify isn't actually "storing" these null and empty lists, but is instead ignoring them which causes them to be null on load.